server/src/main/java/io/crate/planner/operators/SubQueryResults.java

```diff
@@ -29,15 +29,16 @@
 import com.carrotsearch.hppc.ObjectIntMap;
 
 import io.crate.data.Row;
+import io.crate.expression.InputRow;
 import io.crate.expression.symbol.DefaultTraversalSymbolVisitor;
 import io.crate.expression.symbol.OuterColumn;
 import io.crate.expression.symbol.SelectSymbol;
 import io.crate.expression.symbol.Symbol;
 
 public class SubQueryResults {
 
-    public static final SubQueryResults EMPTY = new SubQueryResults(Collections.emptyMap());
     private static final ObjectIntHashMap<OuterColumn> EMPTY_OUTER_COLUMNS = new ObjectIntHashMap<>();
+    public static final SubQueryResults EMPTY = new SubQueryResults(Collections.emptyMap());
```

Comment:
Static fields are initialized in their order. If `SubQueryResults Empty` is created before `ObjectIntHashMap<OuterColumn> EMPTY_OUTER_COLUMNS` and referencing `EMPTY_OUTER_COLUMNS`, the value is null.

See https://docs.oracle.com/javase/specs/jls/se15/html/jls-12.html#jls-12.4.2

Reply:
Maybe also add a comment for this ordering?

Reply:
^^